Fourier log improve message

erh3cq opened the pull request:
During fourier_log deconvolution an error occurs:
Cannot cast ufunc multiply output from dtype('float64') to dtype('int32') with casting rule 'same_kind'.
 This is meant to improve the error message so that people know how to fix the issue.

francisco-dlp commented on hyperspy/_signals/signal1d.py:
```diff
@@ -1207,33 +1207,38 @@ def hanning_taper(self, side='both', channels=None, offset=0):
-                    np.hanning(2 * channels)[-channels:])
-                if offset != 0:
-                    dc[..., -offset:] *= 0.
+        try:
```
What about checking if the dtype is float instead of using the ``try except`` approach?

ericpre committed:
improved message

I have improved the message. Sorry for the long list of commits...I had to reset the branch.

ericpre committed:
Address @francisco-dlp's review comment: check data type and raise a `TypeError` if necessary. Add corresponding tests.

ericpre commented:
This PR is now rebased on RELEASE_next_patch, changed to check the dtype and comes with corresponding tests.
